### PR TITLE
Add description to force-merge tasks

### PR DESCRIPTION
**Pull request by centic9**
This is related to #15975 by providing at least some description for running force-merge tasks and thus allows to see which force-merges on which indices are running and which settings were provided. 

With this PR applied, the Task API will look something like the following for a "GET /_tasks?detailed":

            "bWByk2_lTGKufmq24Inu9g:418" : {
               "node" : "bWByk2_lTGKufmq24Inu9g",
               "action" : "indices:admin/forcemerge",
               "id" : 418,
               "headers" : {},
               "cancellable" : false,
               "running_time_in_nanos" : 161112867379,
               "description" : "Force-merge indices[twitter], maxSegments[-1], onlyExpungeDeletes[false], flush[true]",
               "start_time_in_millis" : 1555624171922,
               "type" : "transport"
            }

This does not provide progress-information, which would be very helpful for long-running merges to see how much longer they will approximately run, but this seems to be non-trivial to add, at least I could not find an easy way (hints for adding this are appreciated if there is a fairly simple approach for doing it!).

### CODE REVIEW

**Comment by martijnvg** [comment]
@elasticmachine test this please

**Comment by martijnvg** [comment]
Thanks @centic9, this is a very useful enhancement.

File: server/src/main/java/org/elasticsearch/action/admin/indices/forcemerge/ForceMergeRequest.java [hunk 1]
```diff
@@ -111,6 +112,14 @@ public ForceMergeRequest flush(boolean flush) {
         return this;
     }
 
+    @Override
+    public String getDescription() {
+        return "Force-merge indices[" + Arrays.toString(indices()) +
```
Comment (martijnvg) [inline server/src/main/java/org/elasticsearch/action/admin/indices/forcemerge/ForceMergeRequest.java]: Arrays.toString(...) already adds brackets, so maybe remove the brackets in string around it?

**Review by martijnvg — COMMENTED** [review]
@centic9 I left one minor comment, do you want to address that?
Otherwise this looks good to me.

File: server/src/main/java/org/elasticsearch/action/admin/indices/forcemerge/ForceMergeRequest.java (hunk 1 above)
Comment (centic9) [reply server/src/main/java/org/elasticsearch/action/admin/indices/forcemerge/ForceMergeRequest.java]: Good catch! Adjusted now.

**Comment by javanna** [comment]
test this please

**Comment by javanna** [comment]
@elasticmachine update branch

**Comment by javanna** [comment]
retest this please

**Comment by javanna** [comment]
Thanks for your contribution and for addressing your PR based on the feedback @centic9 ! One more thing I would do is add a test so that we make sure we don't change this. A unit test that verifies the output of the new toString method would be enough for instance. Otherwise we may end up changing that one day without realizing what effect such change has.

**Comment by centic9** [comment]
Thanks for the response @javanna, I have now added a test which verifies the contents of getDescription() with various different values.

**Comment by martijnvg** [comment]
@elasticmachine test this please

**Comment by martijnvg** [comment]
@centic9 Thanks for updating the PR and apologies for not getting back to you.

**Pull request merged**